src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerConfigPropertyMap.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    public abstract class AnalyzerConfigPropertyMap
+    {
+        public abstract bool TryGetValue(string key, out string value);
+    }
+
+    internal sealed class CompilerAnalyzerConfigPropertyMap : AnalyzerConfigPropertyMap
+    {
+        public static CompilerAnalyzerConfigPropertyMap Empty { get; } = new CompilerAnalyzerConfigPropertyMap(
+            ImmutableDictionary.Create<string, string>(CaseInsensitiveComparison.Comparer));
```

Comment:
config is case insensitive here.  is that documented anywhere?  Is there any way we can encode this into the base class so that all implementations have the right lookup behavior?